andela/Ah-backend-aquaman
# #164069227 enable authenticated authors to follow and unfollow each other

#### What does this PR do?
- Enable authenticated users to follow and unfollow each other
- See a list of a user's followings and followers
#### Description of Task to be completed?
- Add a Follow model and serializer
- Add views for following, unfollowing, get a user’s follows and followers
- Add urls for the views
#### How should this be manually tested?
- Clone the [repo](https://github.com/andela/Ah-backend-aquaman), cd into `Ah-backend-aquaman` and checkout the branch `ft-follow-users-functionality-164069227`
    -  pip install -r requirements.txt
    -  python manage.py migrate
    -  python manage.py runserver
    -  create two accounts and verify your emails
    -  use postman to test the follow endpoints
      - Follow a user: `POST /api/profiles/<username>/follows/`
      - Get a list of followings: `GET /api/profiles/<username>/follows/`
      - Unfollow a user: `DELETE /api/profiles/<username>/follows/`
      - Get a list of followers: `GET /api/profiles/<username>/followers/`
#### What are the relevant pivotal tracker stories?
- [#164069227](https://www.pivotaltracker.com/story/show/164069227)

#### Screenshots (if appropriate)
- Follow a user
<img width="1150" alt="Screenshot 2019-03-18 at 19 09 53" src="https://user-images.githubusercontent.com/6951438/54544876-8c190600-49b1-11e9-992d-d4c5c11a6235.png">
- Get a user's followings
<img width="1214" alt="Screenshot 2019-03-20 at 13 56 20" src="https://user-images.githubusercontent.com/6951438/54679316-06639a80-4b18-11e9-9720-6c75270a9b4f.png">
- Unfollow a user
<img width="1150" alt="Screenshot 2019-03-18 at 19 15 48" src="https://user-images.githubusercontent.com/6951438/54545203-427ceb00-49b2-11e9-9942-fae4a4e24c48.png">
- Get a list of followers
<img width="1214" alt="Screenshot 2019-03-20 at 13 59 41" src="https://user-images.githubusercontent.com/6951438/54679525-8a1d8700-4b18-11e9-8271-5ca1aa9cc4f7.png">
